angular-oauth2-oidc/src/oauth-service.ts

```diff
@@ -600,20 +599,35 @@ export class OAuthService
      */
     public refreshToken(): Promise<object> {
 
+        let params = new HttpParams()
+            .set('grant_type', 'refresh_token')
+            .set('refresh_token', this._storage.getItem('refresh_token'))
+            .set('scope', this.scope);
+        if (this.dummyClientSecret) {
+            params = params.set('client_secret', this.dummyClientSecret);
+        }
+        return this.fetchToken(params);
+    }
+
+    /**
+     * Get token using an intermediate code. Works for the Authorization Code flow.
+     */
+    private getTokenFromCode(code: string): Promise<object> {
+        let params = new HttpParams()
+            .set('grant_type', 'authorization_code')
+            .set('code', code)
+            .set('redirect_uri', window.location.origin);
```

Comment:
this should be `this.redirectUri`

Reply:
@bechhansen this needs to be fixed imo

Reply:
If this is the right spec I believe it specifies redirect_uri and not redirectUri
https://tools.ietf.org/html/rfc6749#section-4.1.1

Reply:
Hi @ceeram 

You are right. I have fixed it.